internal/jujuapi/jimm.go

```diff
@@ -517,3 +521,42 @@ func (r *controllerRoot) Version(ctx context.Context) (apiparams.VersionResponse
 	}
 	return versionInfo, nil
 }
+
+// PrepareModelMigration prepares JIMM for an incoming migration.
+func (r *controllerRoot) PrepareModelMigration(ctx context.Context, args apiparams.PrepareModelMigrationRequest) error {
+	const op = errors.Op("jujuapi.PrepareModelMigration")
+
+	if !r.user.JimmAdmin {
+		return errors.E(op, errors.CodeUnauthorized, "unauthorized")
+	}
+
+	mt, err := names.ParseModelTag(args.ModelTag)
+	if err != nil {
+		return errors.E(op, "invalid model tag", err)
+	}
+
+	if !names.IsValidControllerName(args.TargetControllerName) {
+		return errors.E(op, "invalid controller name")
+	}
+	userMap := map[string]string{}
+	if err := json.Unmarshal([]byte(args.UserMapping), &userMap); err != nil {
+		return errors.E(op, "invalid user map, unable to unmarshal")
+	}
```

Comment:
Would be good not to unmarshal fields from a struct that was already unmarshalled. Indicates our type could be improved.

Reply:
Good point. I think we might actually be able to just use a map[string]string, I'll test now

Reply:
Yeah it worked fine, will update